daemon/snap.go

```diff
@@ -206,6 +228,34 @@ func clientHealthFromHealthstate(h *healthstate.HealthState) *client.SnapHealth
 	}
 }
 
+func isSnapMonitored(st *state.State, instanceName string) (bool, error) {
+	stored := st.Cached("monitored-snaps")
+	if stored == nil {
+		return false, nil
+	}
+	aborts, ok := stored.(map[string]context.CancelFunc)
+	if !ok {
+		return false, fmt.Errorf(`internal error: "monitored-snaps" should be map[string]context.CancelFunc but got %T`, stored)
+	}
+	isMonitored := aborts[instanceName] != nil
+	return isMonitored, nil
+}
+
+func clientSnapRefreshInhibit(st *state.State, snapst *snapstate.SnapState, instanceName string) (*client.SnapRefreshInhibit, error) {
+	monitored, err := isSnapMonitored(st, instanceName)
```

Comment:
can it happen that something is not monitored but has proceed-time set? or would that be just a bug situation?

Reply:
Yes, there is a small window between [continueRefreshOnSnapClose](https://github.com/snapcore/snapd/blob/f7fea7fef0165d764ca3ec6f1d66d1ea7b711501/overlord/snapstate/handlers.go#L1008-L1010) when snap is closed and [doLinkSnap](https://github.com/snapcore/snapd/blob/f7fea7fef0165d764ca3ec6f1d66d1ea7b711501/overlord/snapstate/handlers.go#L2281-L2282) where the snap would not be monitored but still has proceed-time.

Reply:
so it seems to me that the condition here should really be: monitored || proceed-time <= now ?

Reply:
proceed-time isn't the time the snap will be refreshed, it is the time after which the snap will be refreshed in the next auto-refresh (which might be a few hours/minutes after proceed-time). My intention here was to keep it (even when proceed-time <= now) to `snapd-desktop-integration` the opportunity to maybe notify users that a snap will be refreshed soon so they can safely close their open apps.

If this is not the intended behavior, I can add the condition. would love to know your opinion?

Reply:
yes, I misread it, thanks for explaining. I think the stop monitoring + proceed-time <= now is better than checking that proceed-time is reset because it gets reset later in the change so we might falsely report an inhibited snap.

Reply:
yes, I actually misread, sorry for that. updated, thank you!

Reply:
I think the `!monitored || proceed <= now -> means not inhibited` is reasonable because the snap might still be monitored even after proceed-time expires. it might even be a good idea to prune those monitored based on proceed time later.